spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,158 @@
+requires: 2, 6, 10, 23
+required-by: 4
+author: Christopher Goes <cwgoes@tendermint.com>, Juwoon Yun <joon@tendermint.com>
+created: 2019-03-07
+modified: 2019-03-29
+---
+
+## Synopsis
+
+This standards document describes the abstraction of an IBC *connection*: two stateful objects on two separate chains, each tracking the consensus state of the other chain, facilitating cross-chain substate verification. Protocols for safely establishing a connection between two chains, verifying relayed updates (headers) to the consensus state tracked by a connection, cleanly closing a connection, and closing a connection due to detected equivocation are described.
+
+## Specification
+
+### Motivation
+
+The core IBC protocol provides *authorization* and *ordering* semantics for packets: guarantees, respectively, that packets have been committed on the sending blockchain (and according state transitions executed, such as escrowing tokens), and that they have been committed exactly once in a particular order and can be delivered exactly once in that same order. The *connection* abstraction, specified in this standard, defines the *authorization* semantics of IBC (ordering semantics are left to [ICS 4: Channel Semantics](../spec/ics-4-channel-semantics)).
+
+### Desired Properties
+
+- Implementing blockchains should be able to safely allow untrusted actors to open and update connections.
+- The two connecting blockchains should be able to negotiate a shared connection "version". The version consists of metadata about encoding formats that the chain is required to understand, including wire encoding format, accumulator proof format, etc.
+
+#### Pre-Establishment
+
+Prior to connection establishment:
+
+- No further IBC subprotocols should operate, since cross-chain substates cannot be verified.
+- The initiating actor (who creates the connection) must be able to specify a root-of-trust for the chain to connect to and a root of trust for the connecting chain (implicitly, e.g. by sending the transaction).
+
+#### During Handshake
+
+Once a negotiation handshake has begun:
+
+- Only the appropriate handshake datagrams can be executed in order.
+- No chain can masquerade as one of the handshaking chains (formalize...)
+
+#### Post-Establishment
+
+Once a negotiation handshake has completed:
+
+- The created connection objects on both chains contain the roots of trust specified by the initiating actor.
+- No other connection objects can be maliciously created on other chains by replaying datagrams.
+- The connection should be able to be voluntarily & cleanly closed by both blockchains.
+- The connection should be able to be immediately closed upon discovery of a consensus equivocation.
+
+### Technical Specification
+
+#### Definitions
+
+This ICS defines the `Connection` type:
+
+![Datatypes](datatypes.png)
+
+`RootOfTrust`, `Header, and `updateRootOfTrust` are as defined in [ICS 2: Consensus Requirements](../spec/ics-2-consensus-requirements).
```

Comment:
```suggestion
`RootOfTrust`, `Header`, and `updateRootOfTrust` are as defined in [ICS 2: Consensus Requirements](../spec/ics-2-consensus-requirements).
```
(missing `)
Also, the Header and updateRootOfTrust are not known to the reader at this point in the doc. Header is part of ConnTrack struct and updateRootOfTrust() is hidden further down. Maybe add the other data structures here or move the reference where they are introduced.

nit: is the link broken?